cng/rsa.go

```diff
@@ -315,7 +315,7 @@ func newPSS_PADDING_INFO(h crypto.Hash, sizeBits uint32, saltLen int, sign bool)
 
 	// A salt length of -1 and 0 are valid Go sentinel values.
 	if saltLen <= -2 {
-		return info, errors.New("crypto/rsa: PSSOptions.SaltLength cannot be negative")
+		return info, errors.New("crypto/rsa: invalid PSS salt length")
```

Comment:
How about adding the returned salt length to this error message?

Reply:
I would just embed this if else block to following switch 

switch saltLen {
	case -1: // rsa.PSSSaltLengthEqualsHash
		info.Salt = uint32(h.Size())
	case 0: // rsa.PSSSaltLengthAuto
		if sign {
			// Algorithm taken from RFC 3447 Section 9.1.1, which is also implemented by Go at
			// https://github.com/golang/go/blob/54182ff54a687272dd7632c3a963e036ce03cb7c/src/crypto/rsa/pss.go#L288.
			emLen := (sizeBits - 1 + 7) / 8
			hLen := uint32(h.Size())
			info.Salt = emLen - hLen - 2
		} else {
			// Go auto-detects the salt length from the signature structure when verifying.
			// The auto-detection logic is deep in the verification process,
			// we can't replicate it without exhaustive validation.
			err = errors.New("crypto/rsa: rsa.PSSSaltLengthAuto not supported")
		}
	default:
	        // A salt length of -1 and 0 are valid Go sentinel values.
                if saltLen <= -2 {
                        return info, fmt.Errorf("crypto/rsa: invalid PSS salt length: %d", saltLen)
                 }
		info.Salt = uint32(saltLen)
	}

Reply:
We should return the same error message as upstream, else some upstream tests won't pass.